messaging/src/main/java/org/axonframework/serialization/avro/AvroConverter.java

```diff
@@ -0,0 +1,241 @@
+/*
+ * Copyright (c) 2010-2025. Axon Framework
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.axonframework.serialization.avro;
+
+import jakarta.annotation.Nonnull;
+import jakarta.annotation.Nullable;
+import org.apache.avro.generic.GenericRecord;
+import org.apache.avro.message.SchemaStore;
+import org.axonframework.common.infra.ComponentDescriptor;
+import org.axonframework.serialization.ChainingContentTypeConverter;
+import org.axonframework.serialization.ConversionException;
+import org.axonframework.serialization.Converter;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.lang.reflect.Type;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Objects;
+import java.util.function.UnaryOperator;
+
+/**
+ * Converter providing support for <a href="https://avro.apache.org/">Apache Avro</a>, using
+ * <a href="https://avro.apache.org/docs/1.11.0/spec.html#single_object_encoding">Single Object Encoded binary
+ * encoding</a>.
+ * <p>
+ * This converter is intended to work for classes, representing messages specified by Avro Schema.
```

Comment:
Might be nice to add one additional sentence here, specifying this `Converter` implementation **does not** allow conversion purely based on a `Type`. That it is restricted to `Class` is, by the way, entirely fine! But having it in the class-level JavaDoc seems important to me.